src/js/actions/documents.js

```diff
@@ -472,10 +472,14 @@ define(function (require, exports) {
 
                 this.dispatch(events.document.CLOSE_DOCUMENT, payload);
 
-                var newDocument = this.flux.store("application").getCurrentDocument(),
-                    resetLinkedPromise = this.transfer(layerActions.resetLinkedLayers, newDocument),
-                    resetHistoryPromise = newDocument ?
-                        this.transfer(historyActions.queryCurrentHistory, newDocument.id) : Promise.resolve(),
+                return this.flux.store("application").getCurrentDocument();
+            })
+            .tap(function (newDocument) {
```

Comment:
Moved this queryCurrentHistory up in the chain so it will be finished before `resetLinkedLayers` which then does an resetBounds, which amends history.  Same thing in the next diff in this file.